package/cfg/k3s-cis-1.7-hardened/master.yaml

```diff
@@ -151,7 +151,7 @@ groups:
       - id: 1.1.10
         text: "Ensure that the Container Network Interface file ownership is set to root:root (Manual)"
         audit: |
-          ps -ef | grep $kubeletbin | grep -- --cni-conf-dir | sed 's%.*cni-conf-dir[= ]\([^ ]*\).*%\1%' | xargs -I{} find {} -mindepth 1 | xargs --no-run-if-empty stat -c %U:%G
+          ps -ef | grep $kubeletbin | grep -- --cni-conf-dir | echo "/etc/cni/net.d" | sed 's%.*cni-conf-dir[= ]\([^ ]*\).*%\1%' | xargs --no-run-if-empty stat -c %U:%G
```

Comment:
What is the point of
 ``` bash
ps -ef | grep $kubeletbin | grep -- --cni-conf-dir | echo "/etc/cni/net.d"` 
```
`echo` does not use stdin so this statement is equivalent to 
``` bash
echo "/etc/cni/net.d"`
```

The problem here is that $kubectlbin is unset. I am not sure what version of bash is being used but something like this may be what we want
``` bash
ps -ef | kubeletbin=${$kubeletbin:="<DEFAULT_KUBECTL_PATH>"} grep $kubeletbin | grep -- --cni-conf-dir | sed 's%.*cni-conf-dir[= ]\([^ ]*\).*%\1%' | xargs -I{} find {} -mindepth 1 | xargs --no-run-if-empty stat -c %U:%G
````

Reply:
@KevinJoiner the kube-bench variables such as `$kubeletbin` are defined in config.yaml which can be set globally or locally per profile.

Path for "global"/ default : `security-scan/package/cfg/config.yaml`
Path for local: `security-scan/package/cfg/k3s-cis-1.7-permissive/config.yaml`

It would be better to set the variables for each profile's `config.yaml` (locally), instead of hard coding the values in the `audit` comands, to stay align with kube-bench's standards.

For instance in `k3s-cis-1.7-permissive`, the current local config map sets` $kubeletbin` to `containerd`, as shown [here](https://github.com/rancher/security-scan/blob/bb9f870e95d8d709ad090bc2832ba3bc6ee73c8a/package/cfg/k3s-cis-1.7-permissive/config.yaml#L35).
If `containerd` is not found, the other bins might be used by kube-bench as show [here](https://github.com/rancher/security-scan/blob/bb9f870e95d8d709ad090bc2832ba3bc6ee73c8a/package/cfg/config.yaml#L109).

There is an open issue, to go over all possible variables for each distro and assign their values, it would be great if we could fix all these values here.

- https://github.com/rancher/cis-operator/issues/211

As for this specific `audit`, we should test the [original](https://github.com/aquasecurity/kube-bench/blob/d70459b77c28adf7663abbe87458ab109665cb70/cfg/cis-1.7/master.yaml#L133C1-L133C1) one from kube-bench with the right `$kubeletbin` value (set in config.yaml).

Reply:
Sounds good, not 100% sure if we could transfer all the parameters from the global `config.yaml` to the local ones. It should work for all the components variables, though for `target_mapping` and the default variables I would keep the global one intact.

Reply:
Just to clarify, because there's an open issue, I should revert this to the original command for now and the bug will be fixed when https://github.com/rancher/cis-operator/issues/211 gets resolved? Or is there a correct value I should change it to now?